src/main/tut/3.2-adjunctions.md

```diff
@@ -0,0 +1,130 @@
+```Haskell
+swap       :: (a,b) -> (b,a)
+swap (a,b) = (b,a)
+```
+```tut:silent
+def swap[A, B]: ((A, B)) => (B, A) = {
+  case (a, b) => (b, a)
+}
+```
+................
+```Haskell
+return :: d -> m d
+```
+```scala
+def `return`[D]: D => M[D]
+```
+................
+```Haskell
+extract :: w c -> c
+```
+```scala
+def extract[C]: W[C] => C
+```
+................
+```Haskell
+class (Functor f, Representable u) =>
+      Adjunction f u | f -> u, u -> f where
+  unit         :: a -> u (f a)
+  counit       :: f (u a) -> a
+```
+```scala
+abstract class Adjunction[F[_], U[_]](
+    implicit val F: Functor[F],
+    val U: Representable[U]
+){
+  def unit[A](a: => A): U[F[A]]
```

Comment:
why did you decide to make `a` byname here?

Reply:
I don't remember why I've added it, and don't see a reason to keep it.